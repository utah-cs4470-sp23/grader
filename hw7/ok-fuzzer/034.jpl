let a[b] = [argnum, (((if (632 > 888) then (if false then argnum else argnum) else 540) / argnum) + (array[a : 335, b : (sum[a : argnum, b : argnum] argnum), c : (argnum - 69)] (sum[d : a, e : 423] argnum))[{837, 508}{1}, 494, (- 784)])]
assert ([(b != argnum), (b != (sum[c : b] b))][942] && ((if true then {argnum, a} else {argnum, a}){0} != (- (- (b + 267))))), "c"
let c[d, e, f] = (if ((13.7 * (sum[c : argnum, d : b, e : argnum] 99.96)) == (- (- 96.59))) then (array[c : {424}{0}] (array[d : b, e : 691, f : (argnum + 747)] args)) else [(array[c : 527, d : (sum[c : 716] argnum), e : 804] (array[f : b] d)), (array[c : (sum[c : 43] 949), d : b, e : (sum[c : 218, d : argnum] c)] args), (array[c : 378, d : (- argnum), e : b] (array[f : c] argnum))])[239]
write image {(array[g : (array[g : b, h : f] argnum)[(sum[g : b, h : b, i : 415] b), a[d]], h : (sum[g : b] (- g))] {(sum[i : b, j : f] 67.99), (sum[i : d, j : f, k : f] 96.5), 36.69, (sum[i : h] 61.13)}), d}{0} to "g.png"

let g = (((array[g : d] 27.02)[{f}{0}] % (- (if (! true) then (sum[g : e] 92.23) else (sum[g : e, h : d] 49.35)))) == (- (if ((sum[g : 376] g) <= e) then ((- 34.21) % 41.3) else (array[g : e, h : 941] 92.89)[(if false then argnum else f), e])))
let {h, {}} = (if (if ((sum[h : argnum, i : d, j : f] 66.36) <= (sum[h : argnum] 39.37)) then (if (b == f) then (g == false) else (f < d)) else (array[h : e, i : argnum, j : b] g)[f, d, (136 / d)]) then (if (! (if g then false else g)) then (array[h : argnum] {b, {}}) else (array[h : d] {b, {}})) else [(if g then {f, {}} else {e, {}}), (if (! true) then {e, {}} else {e, {}}), {(- e), {}}])[(sum[h : 189] e)]
let i = [((- {argnum}{0}) - (- (- 133)))]
read image "j.png" to j[k, l]

let {} = (if (if (! (919 > f)) then {g}{0} else true) then (array[m : f] (if false then {} else {})) else (if (if g then (h < k) else (b <= f)) then (array[m : (if false then l else f)] (if true then {} else {})) else [{}, (if g then {} else {})]))[{(array[m : k, n : argnum, o : e] o)[argnum, l, 371], argnum, g}{0}]
read image "m.png" to m

let n[o] = (array[n : h] 5.33)
let {p[q, r], s[t], {}} = {[j][argnum], n, {[{}], argnum, false}{0}[(l % h)]}
