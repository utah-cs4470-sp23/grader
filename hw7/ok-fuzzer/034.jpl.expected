(LetCmd
 (ArgLValue (ArrayArgument a b c d))
 (ArrayLoopExpr
  (ArrayType (FloatType) 3)
  a
  (IntExpr (IntType) 184)
  b
  (IntExpr (IntType) 862)
  c
  (UnopExpr
   (IntType)
   -
   (ArrayIndexExpr
    (IntType)
    (IfExpr
     (ArrayType (IntType) 2)
     (FalseExpr (BoolType))
     (ArrayLoopExpr
      (ArrayType (IntType) 2)
      a
      (IntExpr (IntType) 205)
      b
      (VarExpr (IntType) argnum)
      (IntExpr (IntType) 378))
     (ArrayLoopExpr
      (ArrayType (IntType) 2)
      a
      (IntExpr (IntType) 658)
      b
      (IntExpr (IntType) 561)
      (VarExpr (IntType) a)))
    (CallExpr
     (IntType)
     to_int
     (CallExpr (FloatType) tan (FloatExpr (FloatType) 17)))
    (IfExpr
     (IntType)
     (TrueExpr (BoolType))
     (IntExpr (IntType) 257)
     (UnopExpr (IntType) - (IntExpr (IntType) 974)))))
  (FloatExpr (FloatType) 41)))
(LetCmd
 (ArgLValue (VarArgument (FloatType) e))
 (CallExpr
  (FloatType)
  tan
  (CallExpr
   (FloatType)
   acos
   (CallExpr
    (FloatType)
    atan2
    (CallExpr (FloatType) to_float (VarExpr (IntType) c))
    (CallExpr
     (FloatType)
     pow
     (CallExpr (FloatType) to_float (VarExpr (IntType) argnum))
     (FloatExpr (FloatType) 26))))))
(LetCmd
 (ArgLValue (VarArgument (FloatType) f))
 (SumLoopExpr
  (FloatType)
  f
  (VarExpr (IntType) argnum)
  (BinopExpr
   (FloatType)
   (TupleIndexExpr
    (FloatType)
    (TupleLiteralExpr
     (TupleType (FloatType) (ArrayType (FloatType) 3))
     (CallExpr (FloatType) acos (VarExpr (FloatType) e))
     (VarExpr (ArrayType (FloatType) 3) a))
    0)
   -
   (SumLoopExpr
    (FloatType)
    g
    (SumLoopExpr
     (IntType)
     g
     (VarExpr (IntType) b)
     h
     (CallExpr (IntType) to_int (FloatExpr (FloatType) 64))
     i
     (VarExpr (IntType) f)
     (VarExpr (IntType) i))
    (CallExpr (FloatType) to_float (IntExpr (IntType) 597))))))
(LetCmd
 (ArgLValue (ArrayArgument g h i))
 (ArrayLoopExpr
  (ArrayType (FloatType) 2)
  g
  (TupleIndexExpr
   (IntType)
   (TupleLiteralExpr
    (TupleType (IntType))
    (UnopExpr (IntType) - (VarExpr (IntType) d)))
   0)
  h
  (VarExpr (IntType) d)
  (CallExpr
   (FloatType)
   sqrt
   (SumLoopExpr
    (FloatType)
    i
    (ArrayIndexExpr
     (IntType)
     (ArrayLoopExpr
      (ArrayType (IntType) 1)
      i
      (VarExpr (IntType) argnum)
      (IntExpr (IntType) 945))
     (VarExpr (IntType) g))
    (IfExpr
     (FloatType)
     (TrueExpr (BoolType))
     (UnopExpr (FloatType) - (FloatExpr (FloatType) 93))
     (BinopExpr
      (FloatType)
      (FloatExpr (FloatType) 18)
      *
      (FloatExpr (FloatType) 10)))))))
(TimeCmd (ReadCmd "j.png" (ArrayArgument j k l)))
(LetCmd
 (ArgLValue (ArrayArgument m n o p))
 (ArrayIndexExpr
  (ArrayType
   (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
   3)
  (IfExpr
   (ArrayType
    (ArrayType
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     3)
    1)
   (BinopExpr
    (BoolType)
    (IntExpr (IntType) 723)
    >
    (ArrayIndexExpr
     (IntType)
     (ArrayLoopExpr
      (ArrayType (IntType) 2)
      m
      (VarExpr (IntType) argnum)
      n
      (VarExpr (IntType) k)
      (VarExpr (IntType) c))
     (VarExpr (IntType) i)
     (VarExpr (IntType) d)))
   (IfExpr
    (ArrayType
     (ArrayType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      3)
     1)
    (BinopExpr
     (BoolType)
     (TrueExpr (BoolType))
     &&
     (IfExpr
      (BoolType)
      (FalseExpr (BoolType))
      (FalseExpr (BoolType))
      (TrueExpr (BoolType))))
    (ArrayLoopExpr
     (ArrayType
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      1)
     m
     (IfExpr
      (IntType)
      (FalseExpr (BoolType))
      (VarExpr (IntType) k)
      (VarExpr (IntType) argnum))
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      n
      (VarExpr (IntType) c)
      o
      (VarExpr (IntType) m)
      p
      (VarExpr (IntType) argnum)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j)))
    (ArrayLiteralExpr
     (ArrayType
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      1)
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      m
      (VarExpr (IntType) i)
      n
      (VarExpr (IntType) h)
      o
      (VarExpr (IntType) b)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j))
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      m
      (VarExpr (IntType) b)
      n
      (VarExpr (IntType) i)
      o
      (VarExpr (IntType) l)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j))))
   (ArrayLoopExpr
    (ArrayType
     (ArrayType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      3)
     1)
    m
    (IntExpr (IntType) 766)
    (IfExpr
     (ArrayType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      3)
     (BinopExpr
      (BoolType)
      (FloatExpr (FloatType) 59)
      >
      (FloatExpr (FloatType) 49))
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      n
      (IntExpr (IntType) 587)
      o
      (VarExpr (IntType) k)
      p
      (VarExpr (IntType) h)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j))
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       3)
      n
      (VarExpr (IntType) k)
      o
      (VarExpr (IntType) l)
      p
      (VarExpr (IntType) k)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j)))))
  (VarExpr (IntType) b)))
(LetCmd
 (TupleLValue
  (ArgLValue
   (VarArgument
    (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
    q))
  (ArgLValue (ArrayArgument r s t u)))
 (TupleIndexExpr
  (TupleType
   (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
   (ArrayType (FloatType) 3))
  (IfExpr
   (TupleType
    (BoolType)
    (ArrayType (IntType) 1)
    (TupleType
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     (ArrayType (FloatType) 3)))
   (TrueExpr (BoolType))
   (TupleLiteralExpr
    (TupleType
     (BoolType)
     (ArrayType (IntType) 1)
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      (ArrayType (FloatType) 3)))
    (BinopExpr
     (BoolType)
     (UnopExpr (FloatType) - (VarExpr (FloatType) e))
     ==
     (VarExpr (FloatType) f))
    (VarExpr (ArrayType (IntType) 1) args)
    (TupleLiteralExpr
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      (ArrayType (FloatType) 3))
     (VarExpr
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      j)
     (VarExpr (ArrayType (FloatType) 3) a)))
   (IfExpr
    (TupleType
     (BoolType)
     (ArrayType (IntType) 1)
     (TupleType
      (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
      (ArrayType (FloatType) 3)))
    (FalseExpr (BoolType))
    (TupleLiteralExpr
     (TupleType
      (BoolType)
      (ArrayType (IntType) 1)
      (TupleType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       (ArrayType (FloatType) 3)))
     (BinopExpr (BoolType) (FalseExpr (BoolType)) && (FalseExpr (BoolType)))
     (ArrayLiteralExpr
      (ArrayType (IntType) 1)
      (VarExpr (IntType) o)
      (VarExpr (IntType) d))
     (TupleLiteralExpr
      (TupleType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       (ArrayType (FloatType) 3))
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j)
      (VarExpr (ArrayType (FloatType) 3) a)))
    (TupleLiteralExpr
     (TupleType
      (BoolType)
      (ArrayType (IntType) 1)
      (TupleType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       (ArrayType (FloatType) 3)))
     (FalseExpr (BoolType))
     (ArrayLiteralExpr
      (ArrayType (IntType) 1)
      (VarExpr (IntType) i)
      (VarExpr (IntType) k)
      (VarExpr (IntType) b))
     (TupleLiteralExpr
      (TupleType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       (ArrayType (FloatType) 3))
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j)
      (VarExpr (ArrayType (FloatType) 3) a)))))
  2))
(WriteCmd
 (IfExpr
  (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
  (IfExpr
   (BoolType)
   (BinopExpr
    (BoolType)
    (IfExpr
     (IntType)
     (BinopExpr (BoolType) (VarExpr (IntType) u) < (IntExpr (IntType) 954))
     (VarExpr (IntType) argnum)
     (VarExpr (IntType) argnum))
    ==
    (VarExpr (IntType) s))
   (ArrayIndexExpr
    (BoolType)
    (ArrayLoopExpr
     (ArrayType (BoolType) 1)
     v
     (VarExpr (IntType) u)
     (BinopExpr (BoolType) (VarExpr (IntType) o) > (VarExpr (IntType) c)))
    (VarExpr (IntType) c))
   (TupleIndexExpr
    (BoolType)
    (TupleLiteralExpr
     (TupleType (FloatType) (BoolType))
     (FloatExpr (FloatType) 23)
     (UnopExpr (BoolType) ! (FalseExpr (BoolType))))
    1))
  (ArrayLoopExpr
   (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
   v
   (VarExpr (IntType) b)
   w
   (VarExpr (IntType) d)
   (ArrayIndexExpr
    (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
    (VarExpr
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     q)
    (VarExpr (IntType) argnum)
    (VarExpr (IntType) p)))
  (VarExpr
   (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
   q))
 "v.png")
(PrintCmd "v")
(LetCmd
 (TupleLValue (ArgLValue (VarArgument (IntType) v)))
 (ArrayIndexExpr
  (TupleType (IntType))
  (IfExpr
   (ArrayType (TupleType (IntType)) 2)
   (IfExpr
    (BoolType)
    (UnopExpr
     (BoolType)
     !
     (BinopExpr (BoolType) (VarExpr (IntType) d) == (VarExpr (IntType) s)))
    (BinopExpr (BoolType) (VarExpr (IntType) n) < (VarExpr (IntType) d))
    (BinopExpr
     (BoolType)
     (CallExpr (FloatType) sin (FloatExpr (FloatType) 71))
     !=
     (CallExpr (FloatType) tan (FloatExpr (FloatType) 42))))
   (ArrayLoopExpr
    (ArrayType (TupleType (IntType)) 2)
    v
    (UnopExpr (IntType) - (VarExpr (IntType) d))
    w
    (VarExpr (IntType) h)
    (TupleLiteralExpr (TupleType (IntType)) (VarExpr (IntType) n)))
   (ArrayLoopExpr
    (ArrayType (TupleType (IntType)) 2)
    v
    (VarExpr (IntType) k)
    w
    (VarExpr (IntType) d)
    (IfExpr
     (TupleType (IntType))
     (FalseExpr (BoolType))
     (TupleLiteralExpr (TupleType (IntType)) (VarExpr (IntType) d))
     (TupleLiteralExpr (TupleType (IntType)) (VarExpr (IntType) argnum)))))
  (ArrayIndexExpr
   (IntType)
   (IfExpr
    (ArrayType (IntType) 1)
    (BinopExpr
     (BoolType)
     (CallExpr (IntType) to_int (FloatExpr (FloatType) 16))
     <
     (UnopExpr (IntType) - (VarExpr (IntType) d)))
    (ArrayLoopExpr
     (ArrayType (IntType) 1)
     v
     (VarExpr (IntType) n)
     (VarExpr (IntType) n))
    (ArrayLoopExpr
     (ArrayType (IntType) 1)
     v
     (BinopExpr (IntType) (VarExpr (IntType) argnum) * (VarExpr (IntType) k))
     (VarExpr (IntType) o)))
   (VarExpr (IntType) n))
  (VarExpr (IntType) b)))
(LetCmd
 (TupleLValue
  (TupleLValue
   (ArgLValue (VarArgument (IntType) w))
   (ArgLValue (ArrayArgument x y z))))
 (ArrayIndexExpr
  (TupleType
   (TupleType
    (IntType)
    (ArrayType
     (ArrayType (TupleType (FloatType) (FloatType) (FloatType) (FloatType)) 2)
     2)))
  (ArrayLoopExpr
   (ArrayType
    (TupleType
     (TupleType
      (IntType)
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       2)))
    2)
   w
   (VarExpr (IntType) c)
   x
   (VarExpr (IntType) argnum)
   (TupleLiteralExpr
    (TupleType
     (TupleType
      (IntType)
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       2)))
    (TupleLiteralExpr
     (TupleType
      (IntType)
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       2))
     (VarExpr (IntType) p)
     (ArrayLoopExpr
      (ArrayType
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       2)
      y
      (VarExpr (IntType) o)
      z
      (VarExpr (IntType) t)
      (VarExpr
       (ArrayType
        (TupleType (FloatType) (FloatType) (FloatType) (FloatType))
        2)
       j)))))
  (SumLoopExpr
   (IntType)
   w
   (VarExpr (IntType) b)
   x
   (VarExpr (IntType) v)
   y
   (CallExpr
    (IntType)
    to_int
    (CallExpr
     (FloatType)
     sin
     (CallExpr (FloatType) sin (FloatExpr (FloatType) 7))))
   (VarExpr (IntType) u))
  (IntExpr (IntType) 118)))
(TimeCmd
 (LetCmd
  (ArgLValue (ArrayArgument A B C D))
  (ArrayLoopExpr
   (ArrayType (IntType) 3)
   A
   (ArrayIndexExpr
    (IntType)
    (IfExpr
     (ArrayType (IntType) 1)
     (TrueExpr (BoolType))
     (ArrayLiteralExpr (ArrayType (IntType) 1) (VarExpr (IntType) d))
     (ArrayLoopExpr
      (ArrayType (IntType) 1)
      A
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) l)))
    (BinopExpr
     (IntType)
     (VarExpr (IntType) b)
     *
     (BinopExpr (IntType) (VarExpr (IntType) c) / (VarExpr (IntType) w))))
   B
   (VarExpr (IntType) l)
   C
   (VarExpr (IntType) y)
   (TupleIndexExpr
    (IntType)
    (IfExpr
     (TupleType (IntType) (IntType))
     (FalseExpr (BoolType))
     (TupleLiteralExpr
      (TupleType (IntType) (IntType))
      (VarExpr (IntType) i)
      (VarExpr (IntType) s))
     (TupleLiteralExpr
      (TupleType (IntType) (IntType))
      (VarExpr (IntType) A)
      (VarExpr (IntType) l)))
    1))))
